qlib/Mapper.qm

```diff
@@ -947,28 +969,29 @@ Mapper mapv(DataMap);
             if (exists m.constant)
                 v = m.constant;
             else if (exists m.index) {
-                v = do_list ? (map m.index + count + $#, v) : m.index + count;
+                v = do_list ? (map m.index + count + $#, v) : m.index + count; # FIXME iterating empty list
             }
             else if (exists m.runtime) {
                 any rtv = m_runtime{m.runtime};
-                v = do_list ? (map rtv, v) : rtv;
+                v = do_list ? (map rtv, v) : rtv; # FIXME iterating empty list
             }
             else if (m.struct) {
+                # NOTE: hash of lists not supported with struct # FIXME do we ensure it?
                 v = rec;
                 map v = v{m.struct[$1]}, xrange(0, m.struct.size() - 1);
             }
             else
-                v = rec{name} ?? NOTHING;
+                v = rec{name} ?? NOTHING; # mitigate NULL -> NOTHING (?)
```

Comment:
exactly - if you don't mind, I'll keep the comment here, since the difference between NULL and NOTHING needs reminding, at least for me...